### ISSUE TITLE
Runtime Status

### ISSUE THREAD

**Issue by Deniz582** [redhat-developer/vscode-xml#998]
[Error - 2:20:29 PM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 

![image](https://github.com/redhat-developer/vscode-xml/assets/88579040/d9f409d5-f88f-4bf5-8e48-0d7a35cbc617)

**Comment by angelozerr** [comment]
Have you this problem with old version of vscode-xml?

**Comment by Deniz582** [comment]
> Have you this problem with old version of vscode-xml?

no when it reverts back to 0.26.1 it works as intented

**Comment by fbricon** [comment]
can you provide more details about your setup? OS, VS Code version...

**Comment by Deniz582** [comment]
[Error - 2:50:08 PM] Connection to server got closed. Server will not be restarted.
[Error - 2:50:08 PM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 2:50:08 PM] XML Support client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 2:50:08 PM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 

Extension ID: redhat.vscode-xml

**Comment by angelozerr** [comment]
Which version  of vscode are you using?

**Comment by fbricon** [comment]
Do you have a jdk installed? Can you share your user settings.json?

**Comment by fbricon** [comment]
I suspect the binary is already being used, but could be broken or blocked by some antivirus/security. 
@Deniz582 try deleting the [HOME]/.vscode/extensions/redhat.vscode-xml-0.27.0-win32-x64/server/lemminx-win32.exe and try again

Alternatively, try installing a [Java JDK](https://adoptium.net/temurin/releases/?arch=x64&os=windows&package=jdk) and keep `"xml.server.preferBinary": false`

**Comment by phatmn** [comment]
fyi: this error is still here. was trying all the latest versions and the last one that works is v0.26.1
![Image](https://github.com/user-attachments/assets/c90a5ab4-a585-478a-81ca-51612c7b5cf9)
`[error] Error: Pending response rejected since connection got disposed
	at Object.dispose (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:579784)
	at Object.dispose (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:392357)
	at x.handleConnectionClosed (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:392635)
	at x.handleConnectionClosed (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:547086)
	at c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:392445
	at a.invoke (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:581494)
	at o.fire (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:582259)
	at Z (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:568140)
	at a.invoke (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:581494)
	at o.fire (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:582259)
	at h.fireClose (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:590072)
	at Socket.<anonymous> (c:\Users\user\.vscode\extensions\redhat.vscode-xml-0.27.0-win32-x64\dist\extension.js:2:591628)
	at Socket.emit (node:events:531:35)
	at Pipe.<anonymous> (node:net:339:12)`

**Comment by angelozerr** [comment]
@phatmn thanks for your info about 0.26.1 which is working. Are you using binary or Java mode?

**Comment by phatmn** [comment]
@angelozerr java, by default

**Comment by angelozerr** [comment]
Ok and if you try to switch to binary, is it working better?

**Comment by angelozerr** [comment]
Which Java version are you using?

**Comment by phatmn** [comment]
@angelozerr checking "Prefer binary", installing latest version 0.27.2 and restarting extensions doesn't really helped.
I'm using jdk 17.0.3
`IMPLEMENTOR="Oracle Corporation"
JAVA_VERSION="17.0.3"
JAVA_VERSION_DATE="2022-04-19"
LIBC="default"
MODULES="java.base ..."
OS_ARCH="x86_64"
OS_NAME="Windows"
SOURCE=".:git:8d0f7eab8a82 open:git:42f3595dcbe7"`

**Comment by datho7561** [comment]
@phatmn which version of VS Code are you using?

**Comment by datho7561** [comment]
I tried downgrading to VS Code 1.89.1 and I still wasn't able to replicate the bug.

**Comment by phatmn** [comment]
@datho7561 
`Version: 1.95.3 (system setup)
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.22631`

**Comment by rgrunber** [comment]
If I'm understanding correctly that the only thing that works on Windows ARM64 is to install a compatible JDK (or JRE), then that makes sense. We never had support for Windows ARM64. We can only provide binaries of LemMinX for platforms if GitHub providers a corresponding runner. Luckily there's hope with https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/ , but for now it's only available for certain plans.

**Comment by datho7561** [comment]
@albigi I think we should track Windows ARM 64 compatibility as a separate issue, here: #1045

**Comment by xDeniz** [comment]
after installing the `Microsoft.OpenJDK.21` the issue no longer occurs & am using the latest version also thank you @glipscomb 

![Image](https://github.com/user-attachments/assets/ca241772-07e3-4110-9cfc-ea44c7982ea1)

**Comment by glipscomb** [comment]
> [@glipscomb](https://github.com/glipscomb) are you on Windows ARM 64 as well, or are you on Windows x86_64?

Windows x86_64

**Comment by datho7561** [comment]
@glipscomb @xDeniz it's great to hear that using Java works well. If you have some time, could you please try the latest release, 0.28.0, with `"xml.server.preferBinary": true` in your `settings.json`? This forces the binary to be used. If there's something wrong with the binary, I would like to figure it out and fix it. We fixed #1055 , which may have been the root cause of the problem.

**Comment by glipscomb** [comment]
I've set **"xml.server.preferBinary": true**  and all appears to be working as expected

**Comment by datho7561** [comment]
Yay! Okay. I'll close this issue for now, it seems like the root cause was #1055, which has been addressed.
